lib/compat/wordpress-6.9/customizer-preview-custom-css.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Adds inline script to extend customize-preview.js.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Adds JS logic to the Customizer preview for live previewing Custom CSS for Block Themes.
+ *
+ * The logic in this function would be back-ported into customize-preview.js.
+ * This was committed to wordpress-develop trunk in <https://core.trac.wordpress.org/changeset/60522>.
+ */
+function gutenberg_add_customizer_block_theme_custom_css_preview_js() {
+	if ( ! wp_is_block_theme() ) {
+		return;
+	}
+
+	$setting_id = 'custom_css[' . get_stylesheet() . ']';
+
+	$js_function = <<<JS
+		( settingId ) => {
+			wp.customize.bind( 'preview-ready', () => {
+				// Skip running logic that is already merged in trunk.
+				if ( window._wpCustomizeSettings.theme.isBlockTheme ) {
+					return;
+				}
+
+				wp.customize( settingId, function ( setting ) {
+					setting.bind( function ( newValue ) {
+						const style = document.querySelector( 'style#global-styles-inline-css' );
+						if ( ! style ) {
+							return;
+						}
+
+						// Forbid milestone comments from appearing in Custom CSS which would break live preview.
+						newValue = newValue.replace( /\/\*(BEGIN|END)_CUSTOMIZER_CUSTOM_CSS\*\//g, '' );
+
+						style.textContent = style.textContent.replace(
+							/(\/\*BEGIN_CUSTOMIZER_CUSTOM_CSS\*\/)((?:.|\s)*?)(\/\*END_CUSTOMIZER_CUSTOM_CSS\*\/)/,
+							function ( match, beforeComment, oldValue, afterComment ) {
+								return beforeComment + newValue + afterComment;
+							}
+						);
+					} );
+				} );
+			} );
+		}
+JS;
+	wp_add_inline_script(
+		'customize-preview',
+		sprintf( '( %s )( %s )', $js_function, wp_json_encode( $setting_id ) )
+	);
+}
+
+if ( version_compare( get_bloginfo( 'version' ), '6.9.0', '<' ) ) {
+	add_action( 'customize_preview_init', 'gutenberg_add_customizer_block_theme_custom_css_preview_js' );
+}
```

Comment:
This conditional statement will return true if WordPress is in Alpha or Beta, or RC, so the action hook will run even if the JS logic has already been merged into core.

Finally, I would suggest one of the following changes:

1. Remove the conditional statement because there is a skip process [here](https://github.com/WordPress/gutenberg/pull/70428/files?diff=unified#diff-7f78288fbf6f9e1bdd9be4f7b29ae7d499870ba36f754bb6bdaa62ffae01bae1R24-R27).
2. Use `is_wp_version_compatible()`, which allows us to compare with he pure version number:
  ```php
  if ( is_wp_version_compatible( '6.9' ) ) {
  	add_action( 'customize_preview_init', 'gutenberg_add_customizer_block_theme_custom_css_preview_js' );
  }
  ```

Reply:
@t-hamano I included both on purpose actually… I wanted to avoid adding that additional JS when in 6.9 stable. But for 6.9-alpha, I thought it was fine to include it for the sake of devs on different nightly versions.

Reply:
I see, if that's your intention, I think the code is fine as is 👍